### ISSUE TITLE
(When) Does Weblate commit/push changes directly in certain cases?

### ISSUE THREAD

**Issue by BhaaLseN** [WeblateOrg/weblate#1917]
### Steps to reproduce
1. Perform various translation for both the primary language and any other one using one of the filtered results provided by checks (such as: "Strings needing action")
2. When hitting "Save" after a change, notice that the page takes longer to load (doesn't always happen, only sometimes)
3. Notice that a commit and push has been made

### Actual behaviour
Changes were committed upon "Save" and due to configuration also pushed.

### Expected behaviour
I kind-of expected changes to be pending (which is the case for the most part) - until a "Commit" trigger happens (either by cronjob or by using Repository Management > Commit).

But sometimes it would just commit anyways; and I couldn't figure out why this was happening. It was particularly annoying because I wanted to fix a typo in a resource, saved (which made a commit), noticed another typo in the same message, saved (which made another commit) and typo'd while fixing it, so I had to make a 3rd change (which also made another commit) - so a total of 3 commits/pushes within about a minute.
That particular case was like a 100% commit-right-now; but later when I tried to reproduce it I ran through like 20 marked strings and they were correctly entered as pending; to be committed later.
In either case, the repository was clean when I started; no other component in the same project was pending either.

I'm using a cronjob that forces a commit (and push) every hour at the hour (using `./manage.py commit_pending --all --age=0`); and as far as I can tell no other option in the Admin panel (globally, per project nor per component) specifies a different value or indicates immediate commits. The reason for this is to batch changes (rather than getting commits for every change), but at the same time not to leave too much time around since resources still conflict at times due to formatting from translate-toolkit (which is a different problem that needs to be addressed over there; and is something on my TODO-list for a PR).
It was only just now that I noticed this occurring; and the only changes I since was an upgrade from 2.19 to 2.19.1.

### Server configuration
```
$ ./manage.py list_versions
 * Weblate weblate-2.19.1
 * Python 3.5.3
 * Django 2.0.3
 * six 1.11.0
 * social-auth-core 1.7.0
 * social-auth-app-django 2.1.0
 * django-appconf 1.0.2
 * Translate Toolkit 2.2.5
 * Whoosh 2.7.4
 * defusedxml 0.5.0
 * Git 2.11.0
 * Pillow (PIL) 1.1.7
 * dateutil 2.7.0
 * lxml 4.2.0
 * django-crispy-forms 1.7.2
 * compressor 2.2
 * djangorestframework 3.7.7
 * user-agents 1.1.0
 * pytz 2018.3
 * pyuca N/A
 * python-bidi 0.4.0
 * PyYAML 3.12
 * Database backends: django.db.backends.mysql
```

**Comment by nijel** [comment]
See the docs for information when the commit will happen: https://docs.weblate.org/en/latest/admin/continuous.html#lazy-commits

Weblate tries to batch commits, but it also tries to push changes once it makes sense (eg. the translation has been completed).

**Comment by nijel** [comment]
There will be some changes to this soon, see https://github.com/WeblateOrg/weblate/issues/1157

**Comment by BhaaLseN** [comment]
Hm, #1157 sounds like a nice feature; but I'm pretty sure that I was working alone at that time. VCS Updates are also cronjobs (also hourly `./manage.py updategit --all`); so it shouldn't have been an upstream change. I didn't do an import.

But "completion of translation" could be the thing; since one of them was editing the source language (which by definition should be 100% at all times I suppose?)

Perhaps source languages should be excempt from that? Unless I'm misunderstanding that feature, it should trigger when a translation reaches 100% (from <100%), and not when it already was at 100%?

But at least I can now guess _why_ it happened. Although it didn't seem to in the past. I'll continue monitoring this, maybe it is just my imagination and source changes always went straight to VCS.

**Comment by nijel** [comment]
Yes, it only triggered when it was not 100% and after edit it reaches 100%. AFAIK there is no difference in handling source strings. Anyway it would be great if you can figure out circumstances when it you think should not commit.

**Comment by BhaaLseN** [comment]
I just checked again; I have 100% (good/green; not approved/blue because we don't use approvals) on English (source) and German in that component, and for whatever reason it showed me 1 check for the source language. Navigating to that one did not show a check at the side, so I have no idea why it didn't like the string.
![image](https://user-images.githubusercontent.com/826052/37704226-9472b4d0-2cf8-11e8-84c6-80303669fe81.png)

I changed the _German_ translation (not the English one the check brought me to), because I thought it could be improved regardless. Hitting save forced a commit, even though it was at 100% already (both before and after editing; both strings and words according to the component frontpage).

Going from that component, I can always reproduce the immediate commit after a change.
